plan/plan.go

```diff
@@ -175,3 +190,16 @@ func shouldUpdateTTL(desired, current *endpoint.Endpoint) bool {
 	}
 	return desired.RecordTTL != current.RecordTTL
 }
+
+func stripLowerName(dnsName string) (string, error) {
+	dnsName = strings.ToLower(dnsName)
+	dnsName = strings.TrimSpace(dnsName)
+	reg, err := regexp.Compile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
+	if err != nil {
+		return "", err
+	}
+	if !reg.MatchString(dnsName) {
+		return "", fmt.Errorf("%s because dns record is invalid", dnsName)
```

Comment:
nitpick: "dns record" -> "DNS record"

Reply:
also I think the "dns record" itself is not invalid, but the DNS name is not valid (semantically a difference between the "name" and the actual "record")